zeg/http.py

```diff
@@ -41,43 +41,47 @@ def make_session(endpoint, token):
     return session
 
 
+def handle_response(response):
+    if response.status_code == 400:
+        response.reason = response.content.decode("utf-8")
+        response.raise_for_status()
+    if response.content.startswith(b"{"):
```

Comment:
This should just try/catch the `json()` call rather than sort of sniffing content.